Unreal Blackholio Tutorial Documentation (#3253)
clockworklabs/SpacetimeDB: docs/docs/unreal/part-1.md

1. Server (module) code
2. Client code

First, we'll create a project root directory (you can choose the name):

```bash
mkdir blackholio
cd blackholio
```

We'll start by populating the client directory.

## Setting up the Tutorial Unreal Project

In this section, we will guide you through the process of setting up a Unreal Project that will serve as the starting point for our tutorial. By the end of this section, you will have a basic Unreal project and be ready to implement the server functionality.

### Step 1: Create a Blank Unreal Project

SpacetimeDB supports Unreal version `5.6` or later. See [the overview](.) for more information on specific supported versions.

Launch Unreal 5.6 and create a new project by selecting Games from the Unreal Project Browser.

**⚠️ Important: Select the Blank** template and in **Project Defaults select C++**. 

For `Project Name` use `client_unreal`. For Project Location make sure that you use your `blackholio` directory. This is the directory that we created in a previous step.

Click "Create" to generate the blank project.

![Create Blank Project](./part-1-01-create-project.png)

### Import the SpacetimeDB Unreal SDK

> We're working on hosting the SpacetimeDB SDK for Unreal, in the mean time you can download from GitHub:
> > https://github.com/clockworklabs/SpacetimeDB/tree/master/sdks/unreal/src

Reviewer: ```suggestion
While the SpacetimeDB Unreal client SDK is in preview releases, it can only be installed from GitHub:

> https://github.com/clockworklabs/SpacetimeDB/tree/master/sdks/unreal/src

Once the SDK is stabilized, we'll find a more ergonomic way to distribute it.
```

Should also replace the link with one that actually goes directly to a download. I assume this will involve creating a GitHub release, but I'll leave the specifics to you and the Release Ops team.

> REVIEW: We need to decide how we'll share the SDK (eventually we should host with Fab.com and they can add it like any normal Unreal plugin)

Before beginning make sure to close the Unreal project and IDE.

Add the SpacetimeDB Unreal SDK by first adding a new plugin folder:
```bash
cd client_unreal
md Plugins
```

> Copy the SpacetimeDbSdk to the new Plugins folder.
> > You should have /client_unreal/Plugins/SpacetimeDbSdk

Reviewer: ```suggestion
Copy the SpacetimeDbSdk to the new Plugins folder. This should create `/client_unreal/Plugins/SpacetimeDbSdk`.
```

> In the root of the Unreal project right click the client_unreal.uproject and select Generate Visual Studio project files. 

Reviewer: ```suggestion
In the root of the Unreal project, right click the client_unreal.uproject and select **Generate Visual Studio project files**. 
```

![Generate project files](./part-1-02-generate-project.png)

### Create the GameManager Actor

> 1. Open the client_unreal project, you should see a pop-up that requests to build SpacetimeDbSdk, accept that

Reviewer: ```suggestion
1. Open the `client_unreal` project. Unreal will prompt you to build the `SpacetimeDbSdk` module. Do so.
```

Is "module" the correct term here?
